stake-pool/birdbath/src/processor.rs

```diff
@@ -0,0 +1,1100 @@
+            POOL_AUTHORITY_PREFIX,
+            validator_vote_key.as_ref(),
+            &[bump_seed],
+        ];
+        let signers = &[&authority_seeds[..]];
+
+        let ix = spl_token::instruction::burn(
+            token_program.key,
+            burn_account.key,
+            mint.key,
+            authority.key,
+            &[],
+            amount,
+        )?;
+
+        invoke_signed(
+            &ix,
+            &[burn_account, mint, authority, token_program],
+            signers,
+        )
+    }
+
+    #[inline(never)] // needed due to stack size violation
+    fn process_initialize(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let validator_vote_info = next_account_info(account_info_iter)?;
+        let payer_info = next_account_info(account_info_iter)?;
+        let pool_stake_info = next_account_info(account_info_iter)?;
+        let pool_authority_info = next_account_info(account_info_iter)?;
+        let pool_mint_info = next_account_info(account_info_iter)?;
+        let rent_info = next_account_info(account_info_iter)?;
+        let clock_info = next_account_info(account_info_iter)?;
+        let stake_history_info = next_account_info(account_info_iter)?;
+        let stake_config_info = next_account_info(account_info_iter)?;
+        let system_program_info = next_account_info(account_info_iter)?;
+        let token_program_info = next_account_info(account_info_iter)?;
+        let stake_program_info = next_account_info(account_info_iter)?;
+
+        let stake_bump_seed =
+            check_pool_stake_address(program_id, validator_vote_info.key, pool_stake_info.key)?;
+        let authority_bump_seed = check_pool_authority_address(
+            program_id,
+            validator_vote_info.key,
+            pool_authority_info.key,
+        )?;
+        let mint_bump_seed =
+            check_pool_mint_address(program_id, validator_vote_info.key, pool_mint_info.key)?;
+        check_system_program(system_program_info.key)?;
+        check_token_program(token_program_info.key)?;
+        check_stake_program(stake_program_info.key)?;
+
+        let stake_seeds = &[
+            POOL_STAKE_PREFIX,
+            validator_vote_info.key.as_ref(),
+            &[stake_bump_seed],
+        ];
+        let stake_signers = &[&stake_seeds[..]];
+
+        let authority_seeds = &[
+            POOL_AUTHORITY_PREFIX,
+            validator_vote_info.key.as_ref(),
+            &[authority_bump_seed],
+        ];
+        let authority_signers = &[&authority_seeds[..]];
+
+        let mint_seeds = &[
+            POOL_MINT_PREFIX,
+            validator_vote_info.key.as_ref(),
+            &[mint_bump_seed],
+        ];
+        let mint_signers = &[&mint_seeds[..]];
+
+        // change to Rent::get() if i get rid of the invokes that require the AccountInfo
+        let rent = &Rent::from_account_info(rent_info)?;
```

Comment:
We'll need a version of `initialize` on the stake program that doesn't require the rent account... but that can be added!

Reply:
as a style thing, or is there a technical reason? if we want a version that doesnt take rent, id rather just convert `initialize` back to the old style (do the work in instructions then validate in `initialize`). i like this way because its safer but theres no need to have two functions regardless